Packs/Jira/Integrations/JiraV3/JiraV3.py

```diff
@@ -0,0 +1,4381 @@
+from typing import Callable, Tuple
+from collections import defaultdict
+from bs4 import BeautifulSoup
+from datetime import timezone
+import hashlib
+urllib3.disable_warnings()
+# Note: time.time_ns() is used instead of time.time() to avoid the precision loss caused by the float type.
+# Source: https://docs.python.org/3/library/time.html#time.time_ns
+
+""" CONSTANTS """
+JIRA_INCIDENT_TYPE_NAME = 'Jira Incident'
+ISSUE_INCIDENT_FIELDS = {'issue_id': 'The ID of the issue to edit',
+                         'summary': 'The summary of the issue.',
+                         'description': 'The description of the issue.',
+                         'labels': 'A CSV list of labels.',
+                         'components': 'A CSV list of components',
+                         'priority': 'A priority name, for example "High" or "Medium".',
+                         'due_date': 'The due date for the issue (in the format 2018-03-11).',
+                         'assignee': 'The name of the assignee. Relevant for Jira Server only',
+                         'status': 'The name of the status.',
+                         'assignee_id': 'The account ID of the assignee. Use'
+                                        ' the jira-get-id-by-attribute command to get the user\'s Account ID.',
+                         'original_estimate': 'The original estimate of the Jira issue.'
+                         }
+DEFAULT_FETCH_LIMIT = 50
+DEFAULT_FIRST_FETCH_INTERVAL = '3 days'
+DEFAULT_FETCH_INTERVAL = 1  # Unit is in minutes
+DEFAULT_PAGE = 0
+DEFAULT_PAGE_SIZE = 50
+# Errors
+OAUTH2_AUTH_NOT_APPLICABLE_ERROR = ('This command is not applicable to run with a Jira On Prem instance since On Prem instances'
+                                    ' do not use OAuth2 for authorization.')
+ID_OR_KEY_MISSING_ERROR = 'Please provide either the issue ID or key.'
+EPIC_ID_OR_KEY_MISSING_ERROR = 'Please provide either the epic ID or key.'
+CLOSE_INCIDENT_REASON = 'Issue was marked as "Resolved", or status was changed to "Done"'
+MIRROR_DIRECTION_DICT = {
+    'None': None,
+    'Incoming': 'In',
+    'Outgoing': 'Out',
+    'Incoming And Outgoing': 'Both'
+}
+# This will be appended to the attachment's name when mirroring an attachment from XSOAR to Jira
+ATTACHMENT_MIRRORED_FROM_XSOAR = '_mirrored_from_xsoar'
+COMMENT_MIRRORED_FROM_XSOAR = 'Mirrored from Cortex XSOAR'
+
+
+# Exception Classes
+class RevokedAccessTokenError(Exception):
+    """This class is used to raise exceptions when the access token stored in the integration context has been revoked
+    or not valid anymore, so we could try to retrieve another access token using the refresh token or show an error message
+    to the user.
+    """
+
+    def __init__(self):
+        super().__init__(('The access token has been revoked or no authorization was done for your instance, if so,'
+                          ' please refer to the documentation to authorize your instance'))
```

Comment:
Please consult with Shirley about the wording of this err msg.

Reply:
@ShirleyDenkberg Can you please take a look at the wording of this message?

Reply:
(Why if so? I assume this statement is true, so just refer them to the documentation)           
      (What documentation are you referring them to? )  
      
      The access token has been revoked or no authorization was done for your instance.'
                          ' Refer to the documentation to authorize your instance.

Reply:
Good.